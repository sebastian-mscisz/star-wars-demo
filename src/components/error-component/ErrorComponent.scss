@import 'src/variables';

.sw-error-wrapper {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;

    &__content {
        width: 50%;
        height: fit-content;
        max-width: 500px;
        background-color: $base-background-color;
        padding: 45px;
        border: 2px solid $base-color;
        color: black;
        text-shadow:
            -1px -1px 0 #ffee58,
            1px -1px 0 #ffee58,
            -1px 1px 0 #ffee58,
            1px 1px 0 #ffee58;

        &__button {
            width: 85px;
            height: 45px;
            background-color: $base-background-color;
            border: 2px solid $base-color;
            margin-top: 12px;

            & a {
                text-decoration: none;
                color: $base-color;
                transition: all 0.2s linear;

                &:hover {
                    color: $base-color-light;
                }
            }
        }
    }
}
