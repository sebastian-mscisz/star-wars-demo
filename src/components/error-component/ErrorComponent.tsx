import React from 'react';
import { NavLink } from 'react-router-dom';
import { landingPagePath } from 'src/common';
import './ErrorComponent.scss';

type ErrorComponentProps = {
    code: string;
    message: string;
};

export const ErrorComponent: React.FC<ErrorComponentProps> = ({ code, message }) => {
    return (
        <div className="sw-error">
            <h1>
                {code} {message}
            </h1>
            <button className="sw-error__button">
                <NavLink to={landingPagePath()}>Go home</NavLink>
            </button>
        </div>
    );
};
