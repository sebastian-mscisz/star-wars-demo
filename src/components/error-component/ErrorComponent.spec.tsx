import { BrowserRouter } from 'react-router-dom';
import { ErrorComponent } from './ErrorComponent';
import { render } from '@testing-library/react';

describe('Error component', () => {
    it('should render', () => {
        const { asFragment } = render(<ErrorComponent code="404" message="Not found" />, { wrapper: BrowserRouter });
        expect(asFragment()).toMatchSnapshot();
    });
});
