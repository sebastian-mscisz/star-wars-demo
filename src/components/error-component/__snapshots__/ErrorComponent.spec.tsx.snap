// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should render 1`] = `
<DocumentFragment>
  <div
    class="sw-error"
  >
    <h1>
      404 Not found
    </h1>
    <button
      class="sw-error__button"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Go home
      </a>
    </button>
  </div>
</DocumentFragment>
`;
